File: flask_app/app/models/discount_usage.py
from app.models.base import BaseModel, db
from app.constants import EntityType


class DiscountUsage(BaseModel):
    __tablename__ = "discount_usage"

    discount_id = db.Column(db.String(36), nullable=False)
    user_id = db.Column(db.String(36), nullable=False)
    applied_on = db.Column(db.String(36), nullable=False)
    entity = db.Column(db.Enum(EntityType), nullable=False)
    discount_amount = db.Column(db.Float, nullable=False)
    used_at = db.Column(db.DateTime, nullable=False)

    def __repr__(self):
        return f"<DiscountUsage {self.discount_id}>"

File: springboot_api/src/main/java/com/example/membership_api/model/DiscountUsage.java
package com.example.membership_api.model;

import com.example.membership_api.constants.EntityType;
import jakarta.persistence.*;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "discount_usage")
@EqualsAndHashCode(callSuper = true)
public class DiscountUsage extends BaseModel {

    @Column(name = "discount_id", nullable = false, length = 36)
    private String discountId;

    @Column(name = "user_id", nullable = false, length = 36)
    private String userId;

    @Column(name = "applied_on", nullable = false, length = 36)
    private String appliedOn;

    @Enumerated(EnumType.STRING)
    @Column(name = "entity", nullable = false)
    private EntityType entity;

    @Column(name = "discount_amount", nullable = false)
    private Double discountAmount;

    @Column(name = "used_at", nullable = false)
    private LocalDateTime usedAt;

}
